client/v2/tx/tx.go

```diff
@@ -150,7 +176,7 @@ func SimulateTx(ctx client.Context, flagSet *pflag.FlagSet, msgs ...transaction.
 // BroadcastTx attempts to generate, sign and broadcast a transaction with the
 // given set of messages. It will also simulate gas requirements if necessary.
 // It will return an error upon failure.
-func BroadcastTx(clientCtx client.Context, txf Factory, broadcaster broadcast.Broadcaster, msgs ...transaction.Msg) error {
+func BroadcastTx(ctx context.Context, printer *print.Printer, txf Factory, broadcaster broadcast.Broadcaster, skipConfirm bool, msgs ...transaction.Msg) error {
```

Comment:
I'd avoid having a printer, and just make this return the raw bytes, and let the caller print it the way he wants.
This feels weird to have this here.